podcasts/SwiftUI/UnderlineLinkTextView.swift

```diff
@@ -0,0 +1,57 @@
+import SwiftUI
+import SafariServices
+
+struct UnderlineLinkTextView: View {
+    let text: String
+
+    init(_ text: String) {
+        self.text = text
+    }
+
+    var body: some View {
+        // Extract links and add underline to all of them
+        let links = text.matches(for: "(?:__|[*#])|\\[(.*?)\\]\\(.*?\\)")
+        var reimainingText = text
+        var textView = links.count > 0 ? Text("") : Text(text)
+        for (key, link) in links.enumerated() {
+            let explode = reimainingText.components(separatedBy: link)
+            textView = textView + Text(.init(explode.first ?? "")) + Text(.init(link)).underline() + (key == links.count-1 ? Text(.init(explode[safe: 1] ?? "")) : Text(""))
+            reimainingText = explode[safe: 1] ?? ""
+        }
```

Comment:
I think you can replace all of this regex with `AttributedString`'s markdown handling + Runs:

```suggestion
        var attributedString = try! AttributedString(markdown: text)
        attributedString.runs.filter({ $0.link != nil }).forEach({ run in
            attributedString[run.range].underlineStyle = .init(pattern: .solid)
        })

        var textView = Text(attributedString)
```

Reply:
Oh, this is so much better! Thanks!

Done in 72318660a8e1f3debc8000c1c462835cfa6e3177

I did a small change to avoid the force-unwrap.